firestore/integration_test.go

```diff
@@ -3277,27 +3290,55 @@ func TestIntegration_FindNearest(t *testing.T) {
 		h.mustCreate(doc, beans[i])
 	}
 
-	// Query documents with a vector field
-	vectorQuery := collRef.FindNearest(queryField, []float64{1, 2, 3}, 2, DistanceMeasureEuclidean, nil)
+	for _, tc := range []struct {
+		desc         string
+		vq           VectorQuery
+		wantBeans    []coffeeBean
+		wantResField string
+	}{
+		{
+			desc:      "FindNearest without threshold without resultField",
+			vq:        collRef.FindNearest(queryField, []float64{1, 2, 3}, 2, DistanceMeasureEuclidean, nil),
+			wantBeans: beans[:2],
+		},
+		{
+			desc: "FindNearest threshold and resultField",
+			vq: collRef.FindNearest(queryField, []float64{1, 2, 3}, 3, DistanceMeasureEuclidean, &FindNearestOptions{
+				DistanceThreshold:   Ptr[float64](20),
+				DistanceResultField: resultField,
+			}),
+			wantBeans:    beans[:3],
+			wantResField: resultField,
+		},
+	} {
 
-	iter := vectorQuery.Documents(ctx)
-	gotDocs, err := iter.GetAll()
-	if err != nil {
-		t.Fatalf("GetAll: %+v", err)
-	}
+		t.Run(tc.desc, func(t *testing.T) {
+			iter := tc.vq.Documents(ctx)
+			gotDocs, err := iter.GetAll()
+			if err != nil {
+				t.Fatalf("GetAll: %+v", err)
+			}
 
-	if len(gotDocs) != 2 {
-		t.Fatalf("Expected 2 results, got %d", len(gotDocs))
-	}
+			if len(gotDocs) != len(tc.wantBeans) {
+				t.Fatalf("Expected %v results, got %d", len(tc.wantBeans), len(gotDocs))
+			}
 
-	for i, doc := range gotDocs {
-		gotBean := coffeeBean{}
-		err := doc.DataTo(&gotBean)
-		if err != nil {
-			t.Errorf("#%v: DataTo: %+v", doc.Ref.ID, err)
-		}
-		if beans[i].ID != gotBean.ID {
-			t.Errorf("#%v: want: %v, got: %v", i, beans[i].ID, gotBean.ID)
-		}
+			for i, doc := range gotDocs {
+				gotBean := coffeeBean{}
```

Comment:
`var gotBean coffeeBean` is slightly more idiomatic.

Reply:
Done